### PR TITLE
Fix the docstring of xs in pandas/core/generic.py #22892 

### LINKED ISSUE #22892: DOC: Fix the docstring of xs in pandas/core/generic.py
pandas docstring are expected to follow the format defined in https://pandas.pydata.org/pandas-docs/stable/contributing_docstring.html

This includes among other things:
- There is a short description fitting in one line (there can be a longer one after that).
- The description of the parameters, the returns section, and the see also items, starts with a capital letter and ends with a period.
- In the Return section there is no name before the type (e.g. `transformed : Series`) but just the type (e.g. `Series`) in the first line.
- The examples are valid PEP-8 code, and when executed, they produce the shown input.

We should make the required adjustments in the docstring of the function `xs` in `pandas/core/generic.py`.

We should validate the docstring with our validator (which does not validate all the formats, but some):
- `./scripts/validate_docstrings.py pandas.Series.xs`

And we should validate the PEP-8 of the examples (note that the next command will validate the PEP-8 of all examples, just make sure that none of the shown is in the function being addressed, the rest will be fixed in separate issues):
- `flake8 --doctests pandas/core/generic.py`

### PR DESCRIPTION
**Pull request by tlentali**
- [X] closes #22892
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Hello @datapythonista !
As requested in #22892 , I corrected the docstring of the pandas/core/generic.py xs function to pass correctly the following command :

    ./scripts/validate_docstrings.py pandas.Series.xs

And I also validate the PEP-8 according to : 

    flake8 --doctests pandas/core/generic.py

Is it ok for you ?

### CODE REVIEW

**Review by datapythonista — COMMENTED** [review]
Looks good, just couple of comments on things that IMO would make the docstring better.

**Review by datapythonista — COMMENTED** [review]
Looks great. I just added couple of comments about formatting, and about making the examples simpler/clearer (IMO). With them it should be ready to be merged.

Thanks @tlentali

**Comment by tlentali** [comment]
I am on it 👍

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3270,71 +3270,92 @@ class  max_speed
...
+        Defaults to cross-section on the rows (axis=0).
 
         Parameters
         ----------
-        key : object
-            Some label contained in the index, or partially in a MultiIndex
-        axis : int, default 0
-            Axis to retrieve cross-section on
+        key : label
+            Some label contained in the index, or partially in a MultiIndex.
```
[the remark this answers is not in the document]
Comment (tlentali) [reply pandas/core/generic.py]: I am not sure to understand this one, do you want me to precise the key ?

File: pandas/core/generic.py [hunk 2]
```diff
@@ -3270,71 +3270,92 @@ class  max_speed
...
+              num_legs  num_arms  num_wings
+        dog          4         0          0
+        cat          4         0          0
+        duck         2         0          2
+        >>> df.xs('dog')
+        num_legs     4
+        num_arms     0
+        num_wings    0
+        Name: dog, dtype: int64
+        >>> df.xs('num_wings', axis=1)
```
[the remark this answers is not in the document]
Comment (tlentali) [reply pandas/core/generic.py]: Do you want me to display only the second example ?
Thanks for your help @datapythonista !

File: pandas/core/generic.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (datapythonista) [reply pandas/core/generic.py]: I think you can do`df.xs(key=('mammal', 'dog'))`, and if that's the case, the type `label` seems incorrect (or not clear). `label or tuple of label` and an a description that explains what is expected would be better

File: pandas/core/generic.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (datapythonista) [reply pandas/core/generic.py]: I think the best is to move the creation of the multiindex example at the beginning (replacing the first `df`, and keep those examples but with the multiinex dataframe). But if something else makes more sense to you, feel free to do whatever you think it's best. My main point is that `xs` doesn't add much value without multiindexes (and `.loc` would be preferred), and simplifying the examples using only one dataframe would be nice.

**Comment by tlentali** [comment]
I corrected the points you highlighted, is it ok for you ? Again, thanks for you time and your help !

File: pandas/core/generic.py [hunk 3]
```diff
@@ -3270,71 +3270,95 @@ class  max_speed
 
     def xs(self, key, axis=0, level=None, drop_level=True):
         """
-        Returns a cross-section (row(s) or column(s)) from the
-        Series/DataFrame. Defaults to cross-section on the rows (axis=0).
+        Return cross-section from the Series/DataFrame.
+
+        This methode takes a level argument to select data at a particular
```
Comment (datapythonista) [inline pandas/core/generic.py]:
```suggestion
        This method takes a `key` argument to select data at a particular
```

File: pandas/core/generic.py [hunk 4]
```diff
@@ -3270,71 +3270,95 @@ class  max_speed
...
+        level of a MultiIndex easier.
```
Comment (datapythonista) [inline pandas/core/generic.py]:
```suggestion
        level of a MultiIndex.
```

File: pandas/core/generic.py [hunk 5]
```diff
@@ -3270,71 +3270,95 @@ class  max_speed
...
 
         Parameters
         ----------
-        key : object
-            Some label contained in the index, or partially in a MultiIndex
-        axis : int, default 0
-            Axis to retrieve cross-section on
+        key : label or tuple of label, e.g. ['a', 'b', 'c']
```
Comment (datapythonista) [inline pandas/core/generic.py]:
```suggestion
        key : label or tuple of label
```

File: pandas/core/generic.py [hunk 6]
```diff
@@ -3270,71 +3270,95 @@ class  max_speed
...
-        >>> df.xs(('baz', 2), level=[0, 'third'])
-                A  B  C  D
-        second
-        three   5  3  5  3
-
         Returns
         -------
-        xs : Series or DataFrame
+        Series or DataFrame of the cross-section (row(s) or column(s)) from
+        the original Series/DataFrame.
```
Comment (datapythonista) [inline pandas/core/generic.py]: The format of the returns section is:
```
Series or DataFrame  # type or types in a way that they can be parsed (e.g. `bool, int or str` -> `['bool', 'int', 'str']`)
    Cross-section from the original Series or DataFrame corresponding to the selected index levels.
    # ^ Description of what is being returned
```

File: pandas/core/generic.py [hunk 7]
```diff
@@ -3270,71 +3270,95 @@ class  max_speed
...
+        mammal  cat      walks         0
+                dog      walks         0
+                bat      flies         2
+        bird    penguin  walks         2
+        Name: num_wings, dtype: int64
+        >>> df.xs(('mammal', 'dog'))
+                    num_legs  num_wings
+        locomotion
+        walks              4          0
+        >>> df.xs('num_wings', axis=1)
```
Comment (datapythonista) [inline pandas/core/generic.py]: I think we're repeating this, right? See two commands above.

File: pandas/core/generic.py [hunk 8]
```diff
@@ -3270,71 +3270,95 @@ class  max_speed
...
+        class   animal   locomotion
+        mammal  cat      walks         0
+                dog      walks         0
+                bat      flies         2
+        bird    penguin  walks         2
+        Name: num_wings, dtype: int64
+        >>> df.xs(('mammal', 'bat'))
```
Comment (datapythonista) [inline pandas/core/generic.py]: I think this shows the same use case than `df.xs(('mammal', 'dog'))`, I think with just one should be enough.

File: pandas/core/generic.py [hunk 9]
```diff
@@ -3270,71 +3270,95 @@ class  max_speed
...
+                    num_legs  num_wings
+        locomotion
+        flies              2          2
+        >>> df.xs('cat', level=1)
+                           num_legs  num_wings
+        class  locomotion
+        mammal walks              4          0
+        >>> df.xs(('bird', 'walks'), level=[0, 'locomotion'])
```
Comment (datapythonista) [inline pandas/core/generic.py]: Really nice example. And the previous too. Not sure if it'd make it clearer for the readers with examples that returns more than one value? like `df.xs('walks', level=2)` and `df.xs(('mammal', 'walks'), level=[0, 'locomotion'])`

Up to you, whatever you think would help more a newbie to pandas.

**Review by datapythonista — COMMENTED** [review]
Sorry for the late review, the notification for this got lost in my inbox.

Looks great, couple of comments, but almost ready.

Also, it could make sense to leave a blank line between some of the examples, so they are rendered in blocks. If you want to do a `./doc/make.py html --single=pandas.DataFrame.xs`, and see how it looks better (a single code block with all the examples, or couple of code blocks). And if you divide the examples, and you think that it would be helpful to have a comment explaining what we are doing...

**Review by datapythonista — APPROVED** [review]
lgtm, thanks @tlentali

**Comment by tlentali** [comment]
Thank you very much for your help @datapythonista !

**Comment by jreback** [comment]
thanks @tlentali

**Pull request merged**